enhancements/network/netflow.md

```diff
@@ -0,0 +1,132 @@
+
+This proposal discusses how we start providing this functionality by allowing cluster administrators to enable the export of network flow metadata using one of these protocols.
+
+## Motivation
+
+* Cluster administrators need to be able to get network flows out of OpenShift to be consumed by their collectors.
+
+### Goals
+
+- Add support for exporting the network flows traffic in an OpenShift cluster by leveraging the NetFlow/sflow/IPFIX protocols supported by OVS.
+
+### Non-Goals
+
+- This document only applies to OVN-Kubernetes, not OpenShift SDN or any other other network plugin.
+- This document will not discuss the flow logs store solutions. Typically, customers store the NetFlow data in a data warehouse system (e.g. ClickHouse) or search and analytics system (e.g. ElasticSearch).
+  It is out of the scope of this document to discuss which option should be used and if we should manage that solution.
+- This document will not discuss the aggregation of flow data with corresponding K8S pods. This will be discussed in other proposal and analyze various options
+  we can leverage like bundle a collector that aggregate K8S data by querying API on the fly or SkyDive.
+- This document will not discuss flow visualization or flow dashboards.
+
+## Proposal
+
+OVS supports NetFlow v5, IPFIX and sflow as export flows protocol. It also supports a list of collectors by specifying IP and port to send the flow data.
+As such, we would support the three protocols and an arbitrary list of collectors.
+
+The Cluster Network Operator (CNO) would expose:
+* `exportNetworkFlows` optional dict to contain the options to export the network flows
+  * `netflow` optional dict to specify that netflow protocol will be used.
+    * list of strings specifying the IP and port (separated by colon)  of the collectors that will consume the flow data
+  * `sflow` optional dict to specify that sflow protocol will be used.
+    * list of strings specifying the IP and port (separated by colon)  of the collectors that will consume the flow data
+  * `ipfix` optional dict to specify that ipfix protocol will be used.
+    * list of strings specifying the IP and port (separated by colon)  of the collectors that will consume the flow data
+
+This is an example of a *networks.operator.openshift.io* spec exporting flows on all protocols to two collectors on different ports:
+
+```yaml
+spec:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  defaultNetwork:
+    ovnKubernetesConfig:
+      genevePort: 6081
+      mtu: 1400
+    type: OVNKubernetes
+  deployKubeProxy: false
+  disableMultiNetwork: false
+  disableNetworkDiagnostics: false
+  logLevel: Normal
+  managementState: Managed
+  observedConfig: null
+  operatorLogLevel: Normal
+  serviceNetwork:
+  - 172.30.0.0/16
+  exportNetworkFlows:
+    netflow:
+    - 172.30.158.150:2056
+    - 172.30.54.103:2056
+    sflow:
+    - 172.30.158.150:6343
+    - 172.30.54.103:6343
```

Comment:
Above you describe `sflow` as an "optional dict", but here it's an array, not a dict, and in particular, there's nowhere under `sflow` where you could add sampling/polling config. You'd need something like

    sFlow:
      endpoints:
        - 172.30.158.150:2056
        - 172.30.54.103:2056

so you can add other fields later too. (I don't know if "endpoint" is the right terminology here.)

But then, would you necessarily want the same sampling config for every endpoint?

Reply:
yeah, the description needs fixing, it should be an array just like in the example.